### PR TITLE
Upgrade to k8s 0.23 APIs, remove v1beta1 as default, upgrade KEDA

### LINKED ISSUE #4121: Add "idleReplicaCount" in KEDASpec
KEDA has introduced "idleReplicaCount" to support scaling from 0, however Seldon's KedaSpec does not allow that. Can you please add support for that?
KEDA Reference: https://github.com/kedacore/keda/blob/26edd4f551f8f2d6a4d3ddad862c77f757ebac84/apis/keda/v1alpha1/scaledobject_types.go#L69

### PR DESCRIPTION
**Pull request by cliveseldon**
- [x] Upgrade to 0.23 k8s APIs
 - [x] Upgrade KEDA to 2.7.1
 - [x] Change to v1 as only CRD that is created
 - [x] Update licences
 - [x] Remove operator creation of v1beta1 CRD
 - [x] upgrade to allow latest KEDA use

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4121

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | a2a7b9028ecbeed35e06cb2a312665c6010f92f4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-hkkdf) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | a2a7b9028ecbeed35e06cb2a312665c6010f92f4 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-hkkdf) | `/test integration`
notebooks | a70cb3f9499d86c65e290e736d693bf0d2314187 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4136-notebooks-28qfv) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | a70cb3f9499d86c65e290e736d693bf0d2314187 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4136-notebooks-28qfv) | `/test notebooks`
integration | a70cb3f9499d86c65e290e736d693bf0d2314187 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-vcjvv) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | a70cb3f9499d86c65e290e736d693bf0d2314187 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-vcjvv) | `/test integration`
notebooks | b43e006a73e6c283449ccab6388be61aef5f4f38 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4136-notebooks-ktpgc) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: executor/cmd/proxy/main.go [hunk 1]
```diff
@@ -53,13 +53,13 @@ func main() {
 
 	predictor, err := predictor2.GetPredictor(*predictorName, *filename, *sdepName, *namespace, configPath)
 	if err != nil {
-		log.Error(err, "Failed to get predictor")
+		log.Fatal(err, "Failed to get predictor")
 		os.Exit(-1)
```
Comment (axsaucedo) [inline executor/cmd/proxy/main.go:57]: Probably can remove exit as fatal calls exit

File: helm-charts/seldon-core-operator/templates/webhook.yaml [hunk 2]
```diff
@@ -34,9 +20,7 @@ webhooks:
   - v1
   - v1beta1
   clientConfig:
-{{- if not .Values.certManager.enabled }}
```
Comment (axsaucedo) [inline helm-charts/seldon-core-operator/templates/webhook.yaml:37]: Not sure why this keeps getting removed, we should update the script to ensure it's kept, but for now best to revert to keep as is

**Comment by axsaucedo** [comment]
Notebooks tests pass

image

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | b43e006a73e6c283449ccab6388be61aef5f4f38 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-ngc2r) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | b43e006a73e6c283449ccab6388be61aef5f4f38 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4136-integration-2jhr6) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Integration and notebooks tests pass - merging

image

/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4136#issuecomment-1154929725 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**